File: Laboratorios/Lab_8/Livro.cpp
#include "Livro.h"

Livro::Livro(const string &titulo) : titulo(titulo){}

string Livro::getTitulo() const{
    return this->titulo;
}

std::list<Pessoa*> Livro::getAutores() const{
    return this->autores;
}

void Livro::addAutor(Pessoa &autor){
    this->autores.push_back(&autor);
}

void Livro::imprimeLivro(){
    cout << "Titulo: " + this->titulo<< endl;
}

File: Laboratorios/lab_5/aula08-biblioteca-pacotes/projbiblioteca/src/biblioteca/acervo/Livro.java
package biblioteca.acervo;

import java.util.*;

public class Livro
{
    private String titulo;
    private List<Autor> autores = new LinkedList<>();
    private String id;
    
    public Livro(String id, String titulo, Autor primeiroAutor)
    {
        this.id=id;
        this.titulo=titulo;
        autores.add(primeiroAutor);
    }
    public String getTitulo() { return this.titulo; }
    public String getId() { return this.id; }
    public void addAutor(Autor a) { autores.add(a); }

}